controllers/users.js

```diff
@@ -1,7 +1,94 @@
 const User = require('../models/user.js')
 const db = require('../db')
 const bcrypt = require("bcrypt");
-const jwt = require("jsonwebtoken")
+const jwt = require("jsonwebtoken");
+const { findByIdAndUpdate } = require('../models/user.js');
 
 
 db.on('error', console.error.bind(console, 'MongoDB connection error'))
+
+const SALT_ROUNDS = 11;
+const TOKEN_KEY = "securetoken";
+
+//GetUsers
+
+const getUsers = async (req, res)=>{
+    try{
+        const users = await User.find({});
+        return res.status(200).json(users);
+    } catch (error) {
+        res.status(500).json({error: error.message});
+    }
+};
+
+//GetUser
+const getUser = async (req, res) => {
+    try {
+        const { id } = req.params;
+        const user = await User.findById(id);s
+        res.status(200).json(user)
+    } catch (error) {
+        res.status(500).json({error: error.message});
+    }
+};
+
+//UpdateUser
+
+const updateUser = async (req, res) => {
+    try {
+        const { id } = req.params;
+        const updatedUser = await findByIdAndUpdate (id, req.body, {new: true});
+        return res.status(200).json(updatedUser)
+    } catch (error) {
+        return res.status(500).json({error: error.message});
+    }
+};
+
+//DeleteUser
+
+const deleteUser = async (req,res) => {
+    try{
```

Comment:
check the repo for the deleteUser syntax and how they use deletedProduct variable and and if else statement with it
https://git.generalassemb.ly/sei-nyc-ducks/quacking-passwords/blob/master/controllers/products.js

Reply:
done